tests/lib/c_lib/stdio/testcase.yaml

```
@@ -0,0 +1,8 @@
+common:
+  tags:
+    - clib
+  min_ram: 128
+tests:
+  libraries.libc.stdio: {}
+  libraries.libc.stdio.remove:
+    filter: not CONFIG_NATIVE_LIBC
```

This should help us filter-out native_posix in CI for all build configurations, and will also test against multiple libc's.


```suggestion
common:
  tags:
    - clib
  min_ram: 128
  filter: not CONFIG_NATIVE_LIBC
tests:
  libraries.libc.common.stdio.minimal:
    tags: minimal_libc
    filter: CONFIG_MINIMAL_LIBC_SUPPORTED
    extra_configs:
      - CONFIG_MINIMAL_LIBC=y
  libraries.libc.common.stdio.picolibc:
    tags: picolibc
    filter: CONFIG_PICOLIBC_SUPPORTED
    extra_configs:
      - CONFIG_PICOLIBC=y
  libraries.libc.common.stdio.newlib:
    tags: newlib
    filter: TOOLCHAIN_HAS_NEWLIB == 1
    extra_configs:
      - CONFIG_NEWLIB_LIBC=y
```